bus-mapping/src/circuit_input_builder.rs

```diff
@@ -371,20 +371,27 @@ impl CircuitInputBuilder<FixedCParams> {
             self.set_end_chunk(max_rws);
             return;
         }
-
-        // set end block
-        let mut end_block_not_last = self.block.block_steps.end_block_not_last.clone();
-        let mut end_block_last = self.block.block_steps.end_block_last.clone();
-        end_block_not_last.rwc = self.block_ctx.rwc;
-        end_block_last.rwc = self.block_ctx.rwc;
-        end_block_not_last.rwc_inner_chunk = self
-            .chunk_ctx
-            .as_ref()
-            .map_or_else(RWCounter::new, |chunk_ctx| chunk_ctx.rwc);
-        end_block_last.rwc_inner_chunk = self
+        // set padding
+        let last_step = self
+            .block
+            .txs()
+            .last()
+            .map(|tx| tx.last_step())
+            .unwrap_or_else(|| &self.block.block_steps.padding);
+        let mut padding = last_step.clone(); // padding step need to keep transition context
```

Comment:
Heads up: here use last_step because it's at the end of block.
In chunking, we should use next_tx.first_step instead

Reply:
Just to make sure `this_tx.last_step` has the same transition witness as `next_tx.first_step` right?

Reply:
No they are different. I thought we discussed this issue in Pr #1690 review section.